src/fluent/utils.ts

```diff
@@ -18,17 +18,24 @@ const K8S_SA_TOKEN_PATH = "/var/run/secrets/kubernetes.io/serviceaccount/token";
 /**
  * Get the headers for a request
  *
+ * @param token - the token from @kubernetes/client-node
  * @returns the headers for undici
  */
-export async function getHeaders(): Promise<Record<string, string>> {
-  const token = await getToken();
+export async function getHeaders(token?: string | null): Promise<Record<string, string>> {
+  let saToken: string | null = "";
+  if (!token) {
+    saToken = await getToken();
+  }
+
   const headers: Record<string, string> = {
     "Content-Type": "application/json",
     "User-Agent": "kubernetes-fluent-client",
   };
 
   if (token) {
     headers["Authorization"] = `Bearer ${token}`;
+  } else if (saToken) {
+    headers["Authorization"] = `Bearer ${saToken}`;
```

Comment:
```suggestion
    headers["Authorization"] = `Bearer ${svcAcctToken}`;
```
I think svcAcct provides more clarity than sa when reading the function at a glance